src/pq.py

```diff
@@ -0,0 +1,65 @@
+"""Implement a priority queue."""
+
+inf = float("inf")
+
+
+class PikaQ(object):
+    """The Priority Queue Class.
+
+    def __init__(vals=None): Initialize a priority queue, and insert any values passed in into the Priority Queue. Must pass a list of items.
+    def insert(item, pri=float(-inf)): Insert an item into the Priority Queue, if no priority specified, -inf will be the priority,
+    def pop(): Remove the highest priority value from the priority queue.
+    def peek(): Look at next item to be popped in the Priority Queue.
+
+
+    """
+
+    def __init__(self, vals=None):
+        """Initialize a priority queue."""
+        if vals is None:
+            vals = []
+        self.pq = {}
+        if not hasattr(vals, "__iter__"):
+            self.insert(vals)
+        elif hasattr(vals, "__iter__"):
+            if any(isinstance(item, dict) for item in vals):
+                self.insert(vals)
+            else:
+                try:
+                    for val, pri in vals:
+                        self.insert(val, pri)
+                except TypeError:
+                    for val in vals:
+                        self.insert(val)
+                except ValueError:
+                    raise ValueError("Submitting a string? Put it in a list or tuple first, and give it a priority! Otherwise it will be placed in the highest priority group.")
+
+    def insert(self, val, pri=float(-inf)):
```

Comment:
wrapping this in `float` is unnecessary. The `inf` variable that you set is already a float.